### ISSUE TITLE
`XmlMapper`/`UntypedObjectDeserializer` mixes multiple unwrapped collections

### ISSUE THREAD

**Issue by drewko** [FasterXML/jackson-dataformat-xml#445]
Hi I found an issue with deserializing xml node containg multiple unwrapped collections.

**Entity**
```xml
<person>
    <name>a</name>
    <name>b</name>
    <surname>c</surname>
    <surname>d</surname>
</person>
```

**Code**
```java
new XmlMapper().readValue(xml, Object.class);
```

**Output**
```javascript
{
    name = [ c, d ], 
    surname = d
}
```

**Problem**
Could not deserialize node that contains multiple unwrapped collections. If I remove one `surname`, so that `name` is the only collection - it works properly.

Version: 2.12.1

**Comment by cowtowncoder** [comment]
Thank you for reporting this: sounds like a problem indeed. I hope to have a look soon, will add to my todo list.

**Comment by Migwel** [comment]
@cowtowncoder I'm happy to give it a try. To be sure, the expected behavior is that the XML is serialized into
```
{
    name = [ a, b ], 
    surname = [c, d]
}
```
?

**Comment by cowtowncoder** [comment]
@Migwel yes definitely that would be expected. One easy thing to start with would be to write a simple test case, along with ones that exist, under "src/test/java/.../failing" in `jackson-dataformat-xml`. It is possible that the problem could be in `jackson-databind` (there is a little bit of code on databind side to allow for dups),  but should start with XML module.
Fix should be for 2.12, I think, so branch to use would be `2.12`.
